mne/viz/topo.py

```diff
@@ -73,87 +76,335 @@ def iter_topography(info, layout=None, on_pick=None, fig=None,
     """
     import matplotlib.pyplot as plt
 
-    if fig is None:
-        fig = plt.figure()
-
-    fig.set_facecolor(fig_facecolor)
     if layout is None:
-        from ..channels import find_layout
+        from ..channels.layout import find_layout
         layout = find_layout(info)
 
+    if pos is None:
+        pos = layout.pos.copy()
+        if layout_scale is not None:
+            pos[:, :2] *= layout_scale
+
+    if fig is None:
+        fig = plt.figure()
+
     if on_pick is not None:
         callback = partial(_plot_topo_onpick, show_func=on_pick)
         fig.canvas.mpl_connect('button_press_event', callback)
 
+    fig.set_facecolor(fig_facecolor)
+
+    ch_names = _clean_names(info['ch_names'])
+    ch_types = dict((ch_name, channel_type(info, j))
+                    for j, ch_name in enumerate(ch_names))
+
+    for idx, ch_name in enumerate(info["ch_names"]):
+        if ch_name in layout.names:
+            ax = plt.axes(pos[layout.names.index(ch_name)])
+            ch_idx = ch_names.index(ch_name)
+            vars(ax)['_mne_ch_name'] = ch_name
+            vars(ax)['_mne_ch_idx'] = ch_idx
+            vars(ax)['_mne_ch_type'] = ch_types[ch_name]
+            vars(ax)['_mne_ax_face_color'] = str(axis_facecolor)   # why str?
+            yield ax
+
+
+def _scale_layout(layout, layout_scale, ybound=0, zero=0):
+    """Helper function to set scaling factor so subplots don't overlap"""
+
     pos = layout.pos.copy()
-    if layout_scale:
+
+    if layout_scale is 'wide':
+
+        def _make_lines(pos):
```

Comment:
Pleaes add docstring even to private functions. It is helpful for future developers